driver/ni/ai_channels.h

```diff
@@ -0,0 +1,1472 @@
+    if (s == "None") return DAQmx_Val_None;
+    return DAQmx_Val_None;
+}
+
+struct VoltageExcitationConfig {
+    int32_t excit_source;
+    double excit_val;
+    double min_val_for_excitation; // optional
+    double max_val_for_excitation; //optional
+    bool32 use_excit_for_scaling; //optional
+
+    VoltageExcitationConfig(config::Parser &parser)
+        : excit_source(
+              get_excitation_src(parser.required<std::string>("voltage_excit_source"))),
+          excit_val(parser.required<double>("voltage_excit_val")),
+          min_val_for_excitation(parser.optional<double>("min_val_for_excitation", 0)),
+          max_val_for_excitation(parser.optional<double>("max_val_for_excitation", 0)),
+          use_excit_for_scaling(parser.optional<bool32>("use_excit_for_scaling", 0)) {
+    }
+};
+
+struct CurrentExcitationConfig {
+    int32_t excit_source;
+    double excit_val;
+    double min_val_for_excitation; // optional
+    double max_val_for_excitation; //optional
+    bool32 use_excit_for_scaling; //optional
+
+    CurrentExcitationConfig(config::Parser &parser)
+        : excit_source(
+              get_excitation_src(parser.required<std::string>("current_excit_source"))),
+          excit_val(parser.required<double>("current_excit_val")),
+          min_val_for_excitation(parser.optional<double>("min_val_for_excitation", 0)),
+          max_val_for_excitation(parser.optional<double>("max_val_for_excitation", 0)),
+          use_excit_for_scaling(parser.optional<bool32>("use_excit_for_scaling", 0)) {
+    }
+};
+
+
+struct BridgeConfig {
+    int32_t ni_bridge_config;
+    int32_t voltage_excit_source;
+    double voltage_excit_val;
+    double nominal_bridge_resistance;
+
+    BridgeConfig() = default;
+
+    BridgeConfig(config::Parser &parser)
+        : ni_bridge_config(
+              get_bridge_config(parser.required<std::string>("bridge_config"))),
+          voltage_excit_source(
+              get_excitation_src(parser.required<std::string>("voltage_excit_source"))),
+          voltage_excit_val(parser.required<double>("voltage_excit_val")),
+          nominal_bridge_resistance(
+              parser.required<double>("nominal_bridge_resistance")) {
+    }
+};
+
+struct PolynomialConfig {
+    float64 *forward_coeffs;
+    uint32_t num_forward_coeffs;
+    float64 *reverse_coeffs;
+    uint32_t num_reverse_coeffs;
+    int32_t electrical_units;
+    int32_t physical_units;
+
+    PolynomialConfig() = default;
+
+    PolynomialConfig(config::Parser &parser)
+        : num_forward_coeffs(parser.required<uint32_t>("num_forward_coeffs")),
+          num_reverse_coeffs(parser.required<uint32_t>("num_reverse_coeffs")) {
+        auto eu = parser.required<std::string>("electrical_units");
+        auto pu = parser.required<std::string>("physical_units");
+
+        if (ni::UNITS_MAP.find(eu) == ni::UNITS_MAP.end()) {
+            LOG(ERROR) << "Invalid units: " << eu << ". Defaulting to Volts.";
```

Comment:
These types of logs of warning level IMO as it doesn't prevent the task from being configured